### PR TITLE
PLT-1090 Refactored modals that display users

### PR DESCRIPTION
**Pull request by hmhealey**
- Adds UserList and FilteredUserList components based off of the MoreDirectChannels modal
- Uses a FilteredUserList in the MoreDirectChannels, ChannelMembersModal, ChannelInviteModal, and TeamMembersModal

### CODE REVIEW

File: web/react/components/filtered_user_list.jsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            <div
+                className='filtered-user-list'
+                style={this.props.style}
+            >
+                <div className='filter-row'>
+                    <div className='col-sm-6'>
+                        <input
+                            ref='filter'
+                            className='form-control filter-textbox'
+                            placeholder={formatMessage(holders.search)}
```
Comment (crspeller) [inline web/react/components/filtered_user_list.jsx]: Can we user FormattedMessage here to avoid the injectIntl?

Comment (hmhealey) [inline web/react/components/filtered_user_list.jsx]: I actually tried that when I removed the other formatMessage calls from that file, but the placeholder property expects a string. FormattedMessage just produced a placeholder of `[Object object]`

**Comment by jwilander** [comment]
+1

**Comment by crspeller** [comment]
+1

**Pull request merged**